daemon/create.go

```diff
@@ -78,6 +78,21 @@ func (daemon *Daemon) containerCreate(params types.ContainerCreateConfig, manage
 	return containertypes.ContainerCreateCreatedBody{ID: container.ID, Warnings: warnings}, nil
 }
 
+func setContainerIDEnv(container *container.Container, cidenv string) error {
+	if cidenv == "" {
+		return nil
+	}
+	// check if we already have an env var like that
+	for _, env := range container.Config.Env {
+		if strings.HasPrefix(env, cidenv) {
+			return fmt.Errorf("Error environment variable %s already defined", cidenv)
+		}
+	}
+	container.Config.Env = append(container.Config.Env, fmt.Sprintf("%s=%s", cidenv, container.ID))
```

Comment:
When does it become not ok to mutate container struct? Outside of `create` method invocation?

Reply:
We register the container with the engine here:
https://github.com/balena-os/balena-engine/blob/6ac0449ca14cc2871bc498174f9fa891c9437d54/daemon/create.go#L214

So I wouldn't modify it after that...